### PR TITLE
Feature: Add support for meilisearch vectorstore

### LINKED ISSUE #7603: Add support for Meilisearch vector databases
### Feature request

Add support for Meilisearch vector search.
[Meilisearch](https://www.meilisearch.com) is an open-source search engine. See [documentation](https://www.meilisearch.com/docs)

### Motivation

Meilisearch is releasing the vector search/store feature, which should be available from July 31st.

### Your contribution

I'm working on it and will submit a PR for this issue soon.

### PR DESCRIPTION
**Pull request by alallema**
**Description:**

Add support for Meilisearch vector store.
Resolve #7603 

- No external dependencies added
- A notebook has been added

@rlancemartin

https://twitter.com/meilisearch

<!-- Thank you for contributing to LangChain!

Replace this comment with:
  - Description: a description of the change, 
  - Issue: the issue # it fixes (if applicable),
  - Dependencies: any dependencies required for this change,
  - Tag maintainer: for a quicker response, tag the relevant maintainer (see below),
  - Twitter handle: we announce bigger features on Twitter. If your PR gets announced and you'd like a mention, we'll gladly shout you out!

### CODE REVIEW

**Comment by baskaryan** [comment]
@alallema feel free to ping us when it's ready for review

**Comment by alallema** [comment]
Hi @baskaryan and @rlancemartin, my PR is ready 😊 
I've tried to implement the tests as described in the README, but I'm free to modify them if this isn't the right method.

File: libs/langchain/langchain/vectorstores/meilisearch.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        import meilisearch
+    except ImportError:
+        raise ValueError(
+            "Could not import meilisearch python package. "
+            "Please install it with `pip install meilisearch`."
+        )
+    if not client:
+        url = url or get_from_env("url", "MEILI_HTTP_ADDR")
+        try:
+            api_key = get_from_env("api_key", "MEILI_MASTER_KEY")
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/meilisearch.py:31]: should this be
```
api_key = api_key or get_from_env(...)
```

File: libs/langchain/langchain/vectorstores/meilisearch.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+                text_key='text')
+    """
+
+    def __init__(
+        self,
+        embedding: Embeddings,
+        client: Optional[Client] = None,
+        url: Optional[str] = None,
+        api_key: Optional[str] = None,
+        index_name: Optional[str] = "langchain-demo",
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/meilisearch.py:82]: do index_name, text_key and metadata_key need to be Optional[str] or can they just be str

**Review by baskaryan — COMMENTED** [review]
few small comments but mostly looks good!

File: libs/langchain/langchain/vectorstores/meilisearch.py (hunk 2 above)
Comment (alallema) [reply libs/langchain/langchain/vectorstores/meilisearch.py:82]: No, indeed, they have a default value. They will be better in `str`, I will change that

**Comment by baskaryan** [comment]
looks great, thanks @alallema!

**Comment by sabatale** [comment]
@alallema Is there a way to output progress during from_documents(), similarly to https://github.com/meilisearch/compute-embeddings?

**Pull request merged**